project/MimaExcludes.scala

```diff
@@ -83,7 +83,16 @@ object MimaExcludes {
             // SPARK-2757
             ProblemFilters.exclude[IncompatibleResultTypeProblem](
               "org.apache.spark.streaming.flume.sink.SparkAvroCallbackHandler." +
-                "removeAndGetProcessor")
+                "removeAndGetProcessor"),
+            // SPARK-3586
```

Comment:
I think we should not have any MiMa failures here. The Java errors are correct and need to be avoided. I'd have to look again at why the Scala API is changed by this.